app/forms/reactivate_profile_form.rb

```diff
@@ -0,0 +1,65 @@
+class ReactivateProfileForm
+  include ActiveModel::Model
+
+  validates :recovery_code, :password, presence: true
+  validate :validate_password_reset_profile
+  validate :validate_password
+  validate :validate_recovery_code
+
+  attr_accessor :recovery_code, :password
+  attr_reader :user
+
+  def initialize(user, attrs = {})
+    @user = user
+    super attrs
+  end
+
+  def submit(flash)
+    if valid?
+      flash[:recovery_code] = reencrypt_pii
+      true
+    else
+      clear_fields
+      false
+    end
+  end
+
+  protected
+
+  def reencrypt_pii
```

Comment:
I think you can split this up into multiple smaller methods and avoid the reek exclusion. e.g.

```ruby
# can re-use this method in validate_password_reset_profile
def password_reset_profile
  @_password_reset_profile ||= user.password_reset_profile
end

def user_access_key
  @_uak ||= user.unlock_user_access_key(password)
end

def decrypted_pii
  @_pii ||= password_reset_profile.recover_pii(recovery_code)
end

def reencrypt_pii
  recovery_code = password_reset_profile.encrypt_pii(user_access_key, decrypted_pii)
  password_reset_profile.deactivation_reason = nil
  password_reset_profile.save!
  recovery_code
end
```